pkg/image/skopeo.go

```diff
@@ -77,25 +77,32 @@ func (o *skopeoOperations) CopyImage(url, dest, accessKey, secKey string) error
 }
 
 // CopyRegistryImage download image from registry with skopeo
-func CopyRegistryImage(url, dest, accessKey, secKey string) error {
+func CopyRegistryImage(url, dest, destFile, accessKey, secKey string) error {
 	skopeoDest := "dir:" + dest + dataTmpDir
 	err := SkopeoInterface.CopyImage(url, skopeoDest, accessKey, secKey)
 	if err != nil {
 		os.RemoveAll(dest + dataTmpDir)
 		return errors.Wrap(err, "Failed to download from registry")
 	}
-	err = extractImageLayers(dest)
+	err = extractImageLayers(dest, destFile)
 	if err != nil {
 		return errors.Wrap(err, "Failed to extract image layers")
 	}
 
+	//If a specifc file was requested verify it exists, if not - fail
+	if len(destFile) > 0 {
```

Comment:
if len(destFile) > 0 && err != nil {

Otherwise we can potentially overwrite an existing error.

Reply:
if there was an error it was returned in line 89

Reply:
Good point, I missed the return.